enginetest/load_queries.go

```diff
@@ -74,6 +75,110 @@ var LoadDataScripts = []ScriptTest{
+				"asian int, american_indian int, mexican_american int, multi_racial int, other_race int," +
+				"on_probation int, on_parole int, felony int, misdemeanor int, other_offense int," +
+				"convicted_or_sentenced int, detained_or_awaiting_trial int, first_time_incarcerated int, employed int," +
+				"unemployed int, citizen int, noncitizen int, juvenile int, juvenile_male int, juvenile_female int," +
+				"death_row_condemned int, solitary_confinement int, technical_parole_violators int," +
+				"source_url varchar(2043) NOT NULL, source_url_2 varchar(2043), civil_offense int, federal_offense int," +
+				"PRIMARY KEY (id,snapshot_date), KEY id (id));",
+			"LOAD DATA INFILE './testdata/test6.csv' INTO TABLE inmate_population_snapshots " +
+				"FIELDS TERMINATED BY ',' " +
+				"LINES TERMINATED BY '\r\n' " +
+				"IGNORE 1 LINES " +
+				"(federal_offense, misdemeanor, total, detained_or_awaiting_trial, felony, snapshot_date, id, source_url, source_url_2)",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM inmate_population_snapshots",
+				Expected: []sql.Row{
+					{"8946", time.Date(2020, 5, 1, 0, 0, 0, 0, time.UTC), nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, 0, nil, nil, nil, 0, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, "https://www.website.gov", "https://www.website.gov/other.html", nil, nil},
+					{"8976", time.Date(2020, 5, 1, 0, 0, 0, 0, time.UTC), 196, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, 0, 73, nil, nil, 123, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, "https://www.website.gov", "https://www.website.gov/other.html", nil, 0},
+					{"8978", time.Date(2020, 5, 1, 0, 0, 0, 0, time.UTC), 0, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, 0, nil, nil, nil, 0, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, "https://www.website.gov", "https://www.website.gov/other.html", nil, nil},
+					{"8979", time.Date(2020, 5, 1, 0, 0, 0, 0, time.UTC), 71, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, 5, 3, nil, nil, 63, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, "https://www.website.gov", "https://www.website.gov/other.html", nil, 0},
+				},
+			},
+		},
+	},
+	{
+		Name: "LOAD DATA handles non-nil default values",
+		SetUpScript: []string{
+			"CREATE TABLE test1 (pk BIGINT PRIMARY KEY, v1 BIGINT DEFAULT (v2 * 10), v2 BIGINT DEFAULT 5);",
+			"CREATE TABLE test2 (pk BIGINT PRIMARY KEY, v1 BIGINT DEFAULT (v2 * 10), v2 BIGINT DEFAULT 5);",
+			"CREATE TABLE test3 (pk BIGINT PRIMARY KEY, v1 BIGINT DEFAULT (pk * 10), v2 BIGINT DEFAULT (v1 - 1));",
+			"CREATE TABLE test4 (pk BIGINT PRIMARY KEY, v1 BIGINT DEFAULT (pk * 10), v2 BIGINT DEFAULT (v1 - 1));",
+			"LOAD DATA INFILE './testdata/test7.txt' INTO TABLE test1;",
+			"LOAD DATA INFILE './testdata/test7.txt' INTO TABLE test2 (pk);", // The (pk) projection results in a different tree
```

Comment:
Thanks for adding column projections, this is a pretty big gap in coverage.

Could we get at least one test that re-orders the columns from their schema order? I.e. `into table test5 (v2, v1, pk)`